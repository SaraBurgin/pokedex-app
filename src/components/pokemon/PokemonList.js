import React, { Component } from 'react';
import axios from 'axios';

import Styled from 'styled-components';
import PokemonCard from './PokemonCard';

const Button = Styled.button`
  font-weight: bold;
  font-size: 25px;
  border-radius: 10px;
  background-color: #ef5350 !important;
  border: 1px solid white;
  &:hover {
    border: 2px solid grey;
  }
`

export default class PokemonList extends Component {
  constructor() {
    super();
    this.state = {
      url: "https://pokeapi.co/api/v2/pokemon?offset=offset&limit=loadNumber",
      pokemon: null,
      offset: 0,
      loadNumber: 52,
    }
    this.handleClick = this.handleClick.bind(this);   
  }

  getNextOffset() {
    return this.state.offset+this.state.loadNumber;
  }

  handleClick(){
    const newOffset = this.getNextOffset();
    this.setState({offset: newOffset}, () => {
      console.log('Offset: ' + this.state.offset)
      this.getMorePokemon();
    })
  }

  componentDidMount() {
    this.getMorePokemon();
  }

  async getMorePokemon() {
    this.setState({url: "https://pokeapi.co/api/v2/pokemon?offset=" + this.state.offset + "&limit=" + this.state.loadNumber});
    const res = await axios.get(this.state.url);
    this.setState({pokemon: res.data['results'] });
  }

  render() {
    return (
      <>
      <div>
        {this.state.pokemon ? (
        <div className="row">
          {this.state.pokemon.map(pokemon => (
          <PokemonCard key={pokemon.name} name={pokemon.name} url={pokemon.url}/>
          ))}
          </div>
        ) : (<h1>Loading pokemon</h1>
        )}
      </div>
      <Button type="button" className="btn btn-info btn-block" onClick={this.handleClick}>Load More</Button>
      </>
      );
  }
}
