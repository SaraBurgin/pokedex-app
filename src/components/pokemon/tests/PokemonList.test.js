import { render, screen, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import PokemonList from '../PokemonList';
import Button from '../PokemonList';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders pokemonList without crashing', async () => {
  const div = document.createElement('div');
  ReactDOM.render(<PokemonList />, div);
  ReactDOM.unmountComponentAtNode(div);
})

it('renders button correctly', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
})

test('check that load more text is visible in button', () => {
  render(<PokemonList />);
  const linkElement = screen.getByText(/Load More/i);
  expect(linkElement).toBeInTheDocument();
});

it('matches snapshot', () => {
  const tree = renderer.create(<PokemonList />).toJSON();
  expect(tree).toMatchSnapshot();
}); 
// When it creates a tree and whenever I expect to match snapshot it goes and looks at the folder structure and looks for a folder called snapshot, in there there should be a file called PokemonList.snapshot that is created the first time that you run this fx. When you make changes to the component you should update the snapshot so this test does not fail. 

// ReactDOM is part of the React library
// expect is from Jest