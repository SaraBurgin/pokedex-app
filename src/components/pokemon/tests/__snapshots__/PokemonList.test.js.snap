// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div>
    <h1>
      Loading pokemon
    </h1>
  </div>,
  <button
    className="sc-hKgILt GNev btn btn-info btn-block"
    data-testid="button"
    onClick={[Function]}
    type="button"
  >
    Load More
  </button>,
]
`;
